### PR TITLE
Docs for new credential provider package

### PR DESCRIPTION
**Pull request by junshun**
*Description of changes:*
First pass at docs describing the new credential provider package.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/content/en/docs/tasks/packages/credential-provider-package/_index.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        sourceRegistry: public.ecr.aws/eks-anywhere
+        credential:
+          - matchImages:
+            - 783794618700.dkr.ecr.us-west-2.amazonaws.com
+            profile: "default"
+            secretName: aws-secret
+            defaultCacheDuration: "5h"
+    ```
+
+1. Create the secret (if changing the secret)
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/credential-provider-package/_index.md:58]:
```suggestion
1. Create the secret. If you are changing the secret, see [complete configuration options]({{< relref "../../../reference/packagespec/credential-provider-package" >}}) for the format of the secret.
```
I may have misunderstood your meaning, so please check if my change is what you meant.

File: docs/content/en/docs/tasks/packages/credential-provider-package/_index.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   see [complete configuration options]({{< relref "../../../reference/packagespec/credential-provider-package" >}}) for format of secret.
+
+1. Create the namespace (if not installing to eksa-packages)
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/credential-provider-package/_index.md:61]:
```suggestion
1. Create the namespace (if not installing to eksa-packages).
```

File: docs/content/en/docs/tasks/packages/credential-provider-package/_index.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+   eksctl anywhere create packages -f credential-provider-package.yaml
+   ```
+   
+1. Validate the installation
+   ```bash
+   eksctl anywhere get packages --cluster <cluster-name>
+   ```
+   
+## Update
+To update package configuration, update credential-provider-package.yaml file, and run the following command:
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/credential-provider-package/_index.md:78]:
```suggestion
To update package configuration, update credential-provider-package.yaml file and run the following command:
```

File: docs/content/en/docs/tasks/packages/credential-provider-package/_index.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+eksctl anywhere apply package -f credential-provider-package.yaml
+```
+
+## Upgrade
+
+Credential-Provider-Package will automatically be upgraded when a new bundle is activated.
+
+## Uninstall
+
+To uninstall credential-provider-package, simply delete the package
```
Comment (chrisnegus) [inline docs/content/en/docs/tasks/packages/credential-provider-package/_index.md:89]:
```suggestion
To uninstall credential-provider-package, simply delete the package:
```

File: docs/content/en/docs/workshops/packages/credential-provider-package/iam_roles_anywhere.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+description: >
+---
+
+This tutorial demonstrates how to configure the credential provider package to authenticate using [IAM Roles Anywhere](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/introduction.html) to pull from a private AWS Elastic Container Registry (ECR).  
+IAM Roles Anywhere enables workloads outside of AWS to access AWS resources by using X.509 digital certificates to obtain temporary AWS credentials. A trust anchor is used to reference a certificate authority with IAM Roles Anywhere. For this use case the **Kuberentes Cluster CA** can be registered and each kubelet client's x509 cert can be used to authenticate to get temporary AWS credentials.
+
+## Prerequisites 
+1. For setting up the certificate authority later you wil need to obtain your cluster's CA. This can be obtain by
+    ```bash
+    # Assuming CLUSTER_NAME and KUBECONFIG is set
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/packages/credential-provider-package/iam_roles_anywhere.md:15]:
```suggestion
    # Assuming CLUSTER_NAME and KUBECONFIG are set:
```

File: docs/content/en/docs/workshops/packages/credential-provider-package/iam_roles_anywhere.md [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    kubectl get secret -n eksa-system ${CLUSTER_NAME}-ca -o yaml | yq '.data."tls.crt"' | base64 -d
+    ```
+1. A role should be created to allow read access for curated packages. This role can be extended to include private registries that you would also like to pull from in your cluster. A sample policy for curated packages would be.
```
Comment (chrisnegus) [inline docs/content/en/docs/workshops/packages/credential-provider-package/iam_roles_anywhere.md:18]:
```suggestion
1. A role should be created to allow read access for curated packages. This role can be extended to include private registries that you would also like to pull from into your cluster. A sample policy for curated packages would be.
```

**Review by chrisnegus — COMMENTED** [review]
Nice work. I have a few wording suggestions.

**Review by TerryHowe — APPROVED** [review]
/lgtm
/approve

**Pull request merged**